zips/zip-0228.rst

```diff
@@ -208,24 +198,48 @@ The party performing the matching has some Swap Orders :math:`\mathsf{SO}_i`, ea
 The party generates the :math:`\mathsf{bsk}` for the binding signature by summing up the :math:`\mathsf{bsk}_i` values of the Swap Orders that are matched into the Swap Bundle. 
 For example, if Swap Orders :math:`\mathsf{SO}_i` and :math:`\mathsf{SO}_j` are matched, then the :math:`\mathsf{bsk} = \mathsf{bsk}_i + \mathsf{bsk}_j`.
 The consensus check remains the same, using these updated values.
+The binding signature is verified against the value commitments of all Actions in all Action Groups.
 
 Consensus Rule Changes
 ----------------------
 
 The transaction verification becomes a nested loop, iterating over all Action Groups and verifying the Actions they contain. 
-Each spend authorization signature MUST be a valid :math:`\mathsf{SpendAuthSig^{Orchard}}` signature over :math:`\mathsf{AGHash}` using :math:`\mathsf{rk}` as the validating key (That is, the signature is over :math:`\mathsf{AGHash}` instead of :math:`\mathsf{SigHash}`).
+Each spend authorization signature MUST be a valid :math:`\mathsf{SpendAuthSig^{Orchard}}` signature over ``orchard_action_groups_digest`` using :math:`\mathsf{rk}` as the validating key 
+(That is, the signature is over ``orchard_action_groups_digest`` instead of :math:`\mathsf{SigHash}`).
 The time limit also needs to be checked during verification. 
 Specifically, the consensus rule becomes (if any checks fail, the transaction MUST be rejected):
 
     - for all AG in ActionGroups do:
 
         - check that AG.timeLimit <= current block height 
+        - compute ``orchard_action_groups_digest`` as per `T.4a: orchard_action_groups_digest`_.
         - for (i = 0; i < AG.nActionsOrchard; i++):
 
             - check the result of ZKAction.Verify on the proof AG.proofsOrchard[i]
-            - hash[i] = BLAKE2b-256("ZcashActionGroup", vActionGroupsOrchard[i].preAuthActionGroup)
-            - check SpendAuthSig.Validate(AG.vActionsOrchard[i].rk, hash[i], AG.vSpendAuthSigsOrchard[i])
+            - check SpendAuthSig.Validate(AG.vActionsOrchard[i].rk, ``orchard_action_groups_digest``, AG.vSpendAuthSigsOrchard[i])
+
+Fees
+----
+
+Each Swap Order can contain an Action that pays fees to the party performing the matching. 
+This Action, paying fees, in ZEC, to the matcher, uses the matcher's details to generate output notes, since we assume, generally, that the matcher is known by all trading parties.
+In the P2P case the matcher is simply the second party out of the two.
+We distinguish between two types of fees for Asset Swaps, namely Miner Fees and Matcher Fees.
```

Comment:
So, the user can create one action with the fees both to the matcher and the network. Later, the matcher extracts some of it (by attaching a third action group to collect the fees).
There is a freedom for the matcher to try and extract as much as possible. However, if he tries to extract all or too much - the tx will not be mined.
In the P2P scenario, the flow for the users is the same and there is no need for a third action group to extract fees.

Let's adjust all 3 paragraphs to reflect this logic.